src/fluent-plugin-mdsd/lib/fluent/plugin/out_mdsd.rb

```diff
@@ -102,7 +102,7 @@ def initialize(logger)
 
     # Return a two-element array [SchemaId, SchemaString]
     def get_schema_info(record)
-        hashkey = record.keys().sort().join()
+        hashkey = record.keys().join()
```

Comment:
This still ignores the field type. The Schema Key should include both field name AND field type.
Perhaps, instead of "record.keys().join()" use this:
def get_schema_key(record)
	key_str = ""
	record.each { |key, value|
		key_str << key << value.class.name
	}
	return key_str
end